### PR TITLE
REF: Avoid post-processing in blockwise op

### PR DESCRIPTION
**Pull request by jbrockmendel**
Small step towards #34714

cc @TomAugspurger IIRC you implemented split_frames/split_items, which this removes

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -1031,11 +1031,36 @@ def _cython_agg_blocks(
         agg_blocks: List[Block] = []
         new_items: List[np.ndarray] = []
         deleted_items: List[np.ndarray] = []
-        # Some object-dtype blocks might be split into List[Block[T], Block[U]]
-        split_items: List[np.ndarray] = []
-        split_frames: List[DataFrame] = []
 
         no_result = object()
+
+        def cast_result_block(result, block: "Block", how: str) -> "Block":
```
Comment (jreback) [inline pandas/core/groupby/generic.py:1037]: can you put this function elsewhere (module level somewhere) and import, in-line functions make much harder to grok.

Comment (jreback) [reply pandas/core/groupby/generic.py:1037]: this likely actually belongs as a method on block itself no?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/generic.py:1037]: i dont think this merits a Block method.  There will be further refactoring coming up with the goal of having this iteration-over-blocks go through a BlockManager method

Comment (jbrockmendel) [reply pandas/core/groupby/generic.py:1037]: @jreback is this a deal-breaker?  id like to postpone this decision until after implementing the BlockManager method that this/apply/ewm eventually call

Comment (jreback) [reply pandas/core/groupby/generic.py:1037]: ok sounds good

**Review by jreback — APPROVED** [review]

**Pull request merged**